src/components/NewService.vue

```diff
@@ -199,19 +228,19 @@
           })
         }
         if (config.services) {
+          config.services.forEach(function (ser) {
+            if (ser.managed === (managed === 'true')) {
+              linkedServices.push(ser)
+            }
+          })
           if (allServices.length > 0) {
-            config.services.forEach(function (ser) {
-              if (ser.managed === (managed === 'true')) {
-                linkedServices.push(ser)
-                configuration[ser.as] = allServices[0][ser.retrieve]
-              }
-            })
             this.noLinkableServices = false
+            linkedServices.forEach(function (ser) {
+              configuration[ser.as] = allServices[0][ser.retrieve]
```

Comment:
- who's allServices[0] again?
- what's the difference between linked and managed?

Reply:
@taliaga Pardon my variable namings. The allServices[0] assignments is to set a default value in the UI (so the dropdown menu defaults to first available service). As for the linked and managed, Linked Services refer to those environment variables that can be provided in the form of another Service registered in the Project (like a Mongo Replicaset Service being provided to an IoT Agent Service being deployed).